### DISCUSSION TITLE
nex step and ip for try in local instance?

### DISCUSSION THREAD

**Discussion by johnfelipe** [google/model-viewer#3487] [Q&A]
> model-viewer-packages@ bootstrap /home/felipe/model-viewer
	> lerna bootstrap --ci

	lerna notice cli v4.0.0
	lerna info versioning independent
	lerna info ci enabled
	lerna info Bootstrapping 4 packages
	lerna info Installing external dependencies
	lerna info Symlinking packages and binaries
	lerna info lifecycle @google/model-viewer@1.12.0~prepare: @google/model-viewer@1.12.0

	> @google/model-viewer@1.12.0 prepare /home/felipe/model-viewer/packages/model-viewer
	> if [ ! -L './shared-assets' ]; then ln -s ../shared-assets ./shared-assets; fi && ../shared-assets/scripts/fetch-khronos-gltf-samples.sh

	Cloning into '../shared-assets/scripts/../models/glTF-Sample-Models'...
	remote: Enumerating objects: 1809, done.
	remote: Counting objects: 100% (1809/1809), done.
	remote: Compressing objects: 100% (1426/1426), done.
	Receiving objects: 100% (1809/1809), 826.40 MiB | 103.00 KiB/s, done.
	remote: Total 1809 (delta 267), reused 1500 (delta 199), pack-reused 0
	Resolving deltas: 100% (267/267), done.
	Updating files: 100% (1536/1536), done.
	~/model-viewer/packages/shared-assets/models/glTF-Sample-Models ~/model-viewer/packages/model-viewer
	🐶 Fetching Khronos glTF sample models.
	🐕 WARNING: This might result in hundreds of megabytes of data usage!
	HEAD is now at acd355a Merge pull request #351 from KhronosGroup/IridescenceLamp
	🎾 Sample models are now available
	~/model-viewer/packages/model-viewer
	lerna info lifecycle @google/model-viewer-docs@0.0.1~prepare: @google/model-viewer-docs@0.0.1
	lerna info lifecycle @google/model-viewer-render-fidelity-tools@0.0.1~prepare: @google/model-viewer-render-fidelity-tools@0.0.1
	lerna info lifecycle @google/model-viewer-space-opera@0.0.1~prepare: @google/model-viewer-space-opera@0.0.1

	> @google/model-viewer-docs@0.0.1 prepare /home/felipe/model-viewer/packages/modelviewer.dev
	> if [ ! -L './shared-assets' ]; then ln -s ../shared-assets ./shared-assets; fi

	> @google/model-viewer-render-fidelity-tools@0.0.1 prepare /home/felipe/model-viewer/packages/render-fidelity-tools
	> if [ ! -L './shared-assets' ]; then ln -s ../shared-assets ./shared-assets; fi && cp -R node_modules/dspbr-pt/dist/shader test/renderers/dspbr-pt/

	> @google/model-viewer-space-opera@0.0.1 prepare /home/felipe/model-viewer/packages/space-opera
	> if [ ! -L './shared-assets' ]; then ln -s ../shared-assets ./shared-assets; fi

**Comment by samaneh-kazemi** [comment]
Hi @johnfelipe, could you please share more details what you're asking?

**Comment by johnfelipe** [comment]
how can use render fidelity tools in local

**Comment by elalish** [comment]
It sounds like you're trying to install our package and having trouble with several parts (fidelity and editor). Can you please ask some very specific questions? What are you trying to accomplish? What have you done so far? What has worked? What has not?

**Comment by johnfelipe** [comment]
this is all my steps:

git clone https://github.com/google/model-viewer.git

#an hour for complete

cd model-viewer
npm install

	1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
	added 713 packages from 409 contributors and audited 713 packages in 17.053s

	95 packages are looking for funding
	  run `npm fund` for details

	found 1 moderate severity vulnerability
	  run `npm audit fix` to fix them, or `npm audit` for details

npm run bootstrap

	> model-viewer-packages@ bootstrap /home/felipe/model-viewer
	> lerna bootstrap --ci

	lerna notice cli v4.0.0
	lerna info versioning independent
	lerna info ci enabled
	lerna info Bootstrapping 4 packages
	lerna info Installing external dependencies
	lerna info Symlinking packages and binaries
	lerna info lifecycle @google/model-viewer@1.12.0~prepare: @google/model-viewer@1.12.0

	> @google/model-viewer@1.12.0 prepare /home/felipe/model-viewer/packages/model-viewer
	> if [ ! -L './shared-assets' ]; then ln -s ../shared-assets ./shared-assets; fi && ../shared-assets/scripts/fetch-khronos-gltf-samples.sh

	Cloning into '../shared-assets/scripts/../models/glTF-Sample-Models'...
	remote: Enumerating objects: 1809, done.
	remote: Counting objects: 100% (1809/1809), done.
	remote: Compressing objects: 100% (1426/1426), done.
	Receiving objects: 100% (1809/1809), 826.40 MiB | 103.00 KiB/s, done.
	remote: Total 1809 (delta 267), reused 1500 (delta 199), pack-reused 0
	Resolving deltas: 100% (267/267), done.
	Updating files: 100% (1536/1536), done.
	~/model-viewer/packages/shared-assets/models/glTF-Sample-Models ~/model-viewer/packages/model-viewer
	🐶 Fetching Khronos glTF sample models.
	🐕 WARNING: This might result in hundreds of megabytes of data usage!
	HEAD is now at acd355a Merge pull request #351 from KhronosGroup/IridescenceLamp
	🎾 Sample models are now available
	~/model-viewer/packages/model-viewer
	lerna info lifecycle @google/model-viewer-docs@0.0.1~prepare: @google/model-viewer-docs@0.0.1
	lerna info lifecycle @google/model-viewer-render-fidelity-tools@0.0.1~prepare: @google/model-viewer-render-fidelity-tools@0.0.1
	lerna info lifecycle @google/model-viewer-space-opera@0.0.1~prepare: @google/model-viewer-space-opera@0.0.1

	> @google/model-viewer-docs@0.0.1 prepare /home/felipe/model-viewer/packages/modelviewer.dev
	> if [ ! -L './shared-assets' ]; then ln -s ../shared-assets ./shared-assets; fi


	> @google/model-viewer-render-fidelity-tools@0.0.1 prepare /home/felipe/model-viewer/packages/render-fidelity-tools
	> if [ ! -L './shared-assets' ]; then ln -s ../shared-assets ./shared-assets; fi && cp -R node_modules/dspbr-pt/dist/shader test/renderers/dspbr-pt/


	> @google/model-viewer-space-opera@0.0.1 prepare /home/felipe/model-viewer/packages/space-opera
	> if [ ! -L './shared-assets' ]; then ln -s ../shared-assets ./shared-assets; fi



npm run serve



	Starting up http-server, serving ./
	Available on:
	  http://127.0.0.1:8080
	  http://192.168.0.13:8080 
	Hit CTRL-C to stop the server
	

![SNAG-0992](https://user-images.githubusercontent.com/428820/169880452-a1e28bd4-62d8-4675-84e1-e300b357256c.png)

but i want something for drag and drop my models like with editor but faster
how can accomplish that?

**Reply by elalish** [reply]
Our web editor runs on you local machine even though it's a website. I don't think anything will be faster if you run it locally. What slowness are you experiencing?

**Reply by johnfelipe** [reply]
how compile or run editor pls, send me commands or cheatsheet

**Reply by elalish** [reply]
Follow the README: `npm run build`; you already found serve, then click `packages` -> `space-opera` -> `editor`

**Reply by johnfelipe** [reply]
lerna success run Ran npm script 'build' in 4 packages in 88.2s:
	lerna success - @google/model-viewer
	lerna success - @google/model-viewer-docs
	lerna success - @google/model-viewer-render-fidelity-tools
	lerna success - @google/model-viewer-space-opera

http://192.168.0.13:8080/

but not show

**Reply by johnfelipe** [reply]
cd /home/felipe/model-viewer/packages/space-opera
	npm run serve


	Starting up http-server, serving ./
	Available on:
	  http://127.0.0.1:8080
	Hit CTRL-C to stop the server
	open: http://127.0.0.1:8080/editor/

how put 0.0.0.0 not 127.0.0.1

**Reply by johnfelipe** [reply]
HOST=0.0.0.0 npm run dev

either
help

**Comment by johnfelipe** [comment]
http://192.168.0.15:8080/packages/space-opera/editor/

![SNAG-1031](https://user-images.githubusercontent.com/428820/170361126-6252d1d2-e20b-4616-9c67-30def1a3f660.png)

	[2022-05-25T20:25:22.380Z]  "GET /packages/space-opera/editor/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36"
	[2022-05-25T20:25:22.401Z]  "GET /packages/space-opera/node_modules/web-animations-js/web-animations-next-lite.min.js" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36"
	[2022-05-25T20:25:22.402Z]  "GET /packages/space-opera/node_modules/js-beautify/js/lib/beautify-html.js" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36"
	[2022-05-25T20:25:22.404Z]  "GET /packages/space-opera/node_modules/js-beautify/js/lib/beautify-css.js" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36"
	[2022-05-25T20:25:22.404Z]  "GET /packages/space-opera/dist/space-opera.js" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36"
	[2022-05-25T20:25:22.415Z]  "GET /packages/space-opera/dist/space-opera.js" Error (404): "Not found"
	[2022-05-25T20:25:22.442Z]  "GET /packages/shared-assets/icons/favicon.png" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36"
	[2022-05-25T20:25:22.485Z]  "GET /packages/shared-assets/icons/favicon.png" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36"

how solve this?

**Reply by johnfelipe** [reply]
@elalish i need help with this, pls give me a hand

**Reply by elalish** [reply]
I would love to, but I still can't figure out what you're trying to do or why. Please answer my questions and give me some background so I can help.

**Comment by designbyadrian** [comment]
I wish to add a feature to a forked version of model-viewer, so I was hoping to be able to run some kind of dev server as I'm developing.

I have run npm install and bootstrap. When running `npm run serve`, all I'm presented with is the directory structure. model-viewer.dev and space-opera both give me incomplete sites with various 404s for javascript files.

_Edit: I forgot to run `nom run build`! Probably confused it with bootstrap._
